src/backlinks/handler.js

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {
+  internalServerError, noContent, notFound, ok,
+} from '@adobe/spacecat-shared-http-utils';
+import AhrefsAPIClient from '../support/ahrefs-client.js';
+import { retrieveSiteBySiteId } from '../utils/data-access.js';
+
+export default async function auditBrokenBacklinks(message, context) {
+  const { type, url: siteId, auditContext } = message;
+  const { dataAccess, log, sqs } = context;
+  const {
+    AUDIT_RESULTS_QUEUE_URL: queueUrl,
+  } = context.env;
+
+  try {
+    log.info(`Received ${type} audit request for siteId: ${siteId}`);
+
+    const site = await retrieveSiteBySiteId(dataAccess, siteId, log);
+    if (!site) {
+      return notFound('Site not found');
+    }
+
+    const auditConfig = site.getAuditConfig();
+    if (auditConfig.auditsDisabled()) {
+      log.info(`Audits disabled for site ${siteId}`);
+      return ok();
+    }
+
+    if (auditConfig.getAuditTypeConfig(type)?.disabled()) {
+      log.info(`Audit type ${type} disabled for site ${siteId}`);
+      return ok();
+    }
+
+    const ahrefsAPIClient = AhrefsAPIClient.createFrom(context);
+    const finalURL = `${site.getBaseURL().split('://')[0]}://www.${site.getBaseURL().split('://')[1]}`;
```

Comment:
see commit message: `chore: for the sake of testing`

Reply:
@solaris007 , @ekremney , if I use the following:
```
const urls = [site.getBaseURL(), toggleWWW(site.getBaseURL())];
```
Then, in terms of persisting the result and sending the message to the post-processor, should it be like there were 2 audits running? Since there would be 2 different references from ahrefs and result sets.

Reply:
@iuliag, this would be fine for apex domains. However, this wouldn't work for urls with subdomains (ie, business.adobe.com).

it's perfectly valid for running the same logic for different urls and adding results to `auditResults` array. In fact, that's what we are doing in apex audit: we check two urls, then add the results in an array to be received by post-processor.

you could do the same. Post-processor may combine the broken backlinks for both www and non-www versions of a url - which I believe would be a very accurate thing to do.

check the apex audit results sent from audit worker to post processor: https://github.com/adobe/spacecat-audit-post-processor/blob/main/test/fixtures/apex-audit-results.js

Reply:
That's another approach to it.
However:
1. we get the broken backlinks sorted from top most important to less, and the concat wouldn't know how to merge.
2. Link in fullAuditRef would be different depending on target of the query, so we'd still need to store 2 audits.

Reply:
in LHS i run desktop and mobile strategies as separate audit types (lhs-mobile vs lhs-desktop). but you can also run within the same audit and have a multi-auditResult. don't get hung up on the fullAuditRef

Reply:
I opted to go with 2 audits in the end. Please re-review.